src/main/java/com/es/phoneshop/service/ProductService.java

```diff
@@ -10,8 +10,6 @@ public interface ProductService {
 
     Product getProduct(Long id);
 
-    Product getProductByCode(String code);
-
     List<Product> findProducts();
```

Comment:
it's still not used
let's move the check "if (StringUtils.isBlank(description))" from DAO to the servlet method and call the necessary method in service, because when I'm looking to the servlet it seems that we are searching by description always, and only when I move to the DAO layer I see that we don't execute search in case of blank strings
it's much more clear when developer divides the method execution as early as it's possible